client-app/shared/layout/components/search-bar/search-bar.vue

```diff
@@ -10,16 +10,15 @@
     <div class="flex flex-grow relative">
       <input
         v-model.trim="searchPhrase"
-        :disabled="loading"
         :placeholder="$t('shared.layout.search_bar.enter_keyword_placeholder')"
         maxlength="30"
         class="flex-grow mr-4 rounded h-10 px-4 font-medium text-sm outline-none disabled:bg-gray-200"
-        @keyup.enter="search"
+        @keyup.enter="onInputChange"
         @keyup.esc="searchDropdownVisible && hideSearchDropdown()"
-        @input="search"
+        @input="onInputChange"
       />
 
-      <VcButton class="uppercase px-4 !h-10" @click="search">
+      <VcButton class="uppercase px-4 !h-10" @click="searchPhrase.length > 0 && onSearchButtonClick">
```

Comment:
move condition logic to method

Reply:
We have same type of logic throughout this component. It's just a precondition to call it

Reply:
Better encapsulate all logic inside the method.